optimum/habana/transformers/generation/utils.py

```diff
@@ -681,8 +647,16 @@ def generate(
                     "Please refer to the documentation for more information. "
                     "(https://huggingface.co/docs/transformers/main/en/main_classes/text_generation)"
                 )
-            generation_config.max_length = generation_config.max_new_tokens + input_ids_length
-        self._validate_generated_length(generation_config, input_ids_length, has_default_max_length)
+            if "token_idx" in model_kwargs:
+                generation_config.max_length = input_ids_length
+            else:
+                generation_config.max_length = generation_config.max_new_tokens + input_ids_length
+
+        self._validate_generated_length(
+            generation_config,
+            model_kwargs["token_idx"] if "token_idx" in model_kwargs else input_ids_length,
```

Comment:
```suggestion
            model_kwargs["token_idx"].item() if "token_idx" in model_kwargs else input_ids_length,
```
This fixes it on my side. Can you try it and see if that solves your error?

Reply:
my gaudi2 image and firmware are 1.13 since my system is ubuntu 20.04, and I have some issue to upgrade to 22.04. 1.14 does only support ubuntu 20.04.I think this is why my error is different with you

Reply:
Ah yes that must be related then!